### PR TITLE
uuid documentation update

### PR DESCRIPTION
**Pull request by burner**
* moved some examples from comments to unittests
* one ctor was disabled, reimplemented it

### CODE REVIEW

File: std/uuid.d [hunk 1]
```diff
@@ -256,25 +259,28 @@ public struct UUID
             assert(ctfe.data == data);
         }
 
-/+
-        Not Working! DMD interprets the ubyte literals as ints, then complains the int can't
-        be converted to ubyte!
```
Comment (quickfur) [inline std/uuid.d]: Did you add a unittest to make sure this particular case works now?

**Comment by quickfur** [comment]
Other than that, LGTM. Thanks!

File: std/uuid.d (hunk 1 above)
Comment (burner) [inline std/uuid.d]: no but, I added an template restriction that only allows integrals. Inside the ctor I use the "to" template to convert them to ubyte. So every int type passed in whose value fits into a ubyte will work.

Comment (quickfur) [inline std/uuid.d]: Ah, I see. That makes sense.

**Comment by quickfur** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks

**Pull request merged**